CMakeLists.txt

```diff
@@ -296,6 +296,16 @@ set (CMAKE_C_FLAGS                       "${CMAKE_C_FLAGS} ${COMPILER_FLAGS} ${C
 set (CMAKE_C_FLAGS_RELWITHDEBINFO        "${CMAKE_C_FLAGS_RELWITHDEBINFO} -O3 ${CMAKE_C_FLAGS_ADD}")
 set (CMAKE_C_FLAGS_DEBUG                 "${CMAKE_C_FLAGS_DEBUG} -O0 -g3 -ggdb3 -fno-inline ${CMAKE_C_FLAGS_ADD}")
 
+# On platforms other than macOS, linkers have an option to disable size optimizations. (-O0)
+# By setting the options, builds (and in particular, linking stages) may be marginally faster.
+# Additionally, the linker behavior will fall closer in line with what the user would expect.
+# For releases, we allow the linker optimizer to do whatever it can, so options are left as-is.
+if (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG")
+    if (NOT OS_DARWIN)
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-O0")
+    endif()
+endif()
+
```

Comment:
I would consider adding it as a separate flag e.g. DISABLE_LINKER_OPTIMISATIONS and adding the log message as the Debug build is extremely common among developers (and this PR implicitly changes the default behaviour with no notification), e.g.

```cmake
if (DISABLE_LINKER_OPTIMIZATIONS)
 # do stuff
 # warning so users could grep and see the non-default (non-expected) behaviour
 message(WARNING "Disabling linker optimizations")
else()
 message(STATUS "Enabling linker optimizations")
endif()
```

Reply:
That is very nearly what the PR did, before it was implied that this should be the default for debug builds. Should I be changing it back?

Reply:
It's your choice whether to turn this on or off by default, I just say that this option should also be added as a flag (so one could switch it if needed).